### ISSUE TITLE
volume rendering

### ISSUE THREAD

**Issue by xmatakt** [xmatakt/bakalarka#6]
Tak az dnes sa mi podarilo dostat k tomu volume renderingu, posledne dni som si iba po veceroch cital co to je a hladal zdroje.
To co som sem pridal je vlastne iba prepis toho co ste mi poslali link do reci OpenTK. 
Dalej sa budem snazit pochopit kod, popripade ho vylepsit, pretoze vysledok nieje nijako ohurujuci, ale za zaklad by sa to snad mohlo povazovat.

**Comment by RobertSpir** [comment]
doplnte tam aj shaders projekt, lebo mi to nefunguje so starym shaders.dll

**Comment by RobertSpir** [comment]
este tff.dat chyba

**Comment by RobertSpir** [comment]
uz som si ho nasiel, nahodou to vyzera celkom zaujimavo

**Comment by xmatakt** [comment]
Tak, dnes som spravil podla mna krajsi aj ked nefarebny rendering.
Data, ktore chcete zobrazit, treba vybrat vo Volume.cs (su tam jednotlive dostupne zakomentovane).
Dalej treba (momentalne nie, ale keby ste prepinali medzi shaderami (tiez comment + uncomment)) aj stepSize.
Tu je zdroj: http://graphicsrunner.blogspot.sk/2009/01/volume-rendering-101.html.
Vecer by som chcel spravit volbu stepSize pre uzivatela, vyber dat (k tomu budem musiet asi spravit dalsie textkaky s rozmermi dat, tusim aj do ParaView bolo treba zadavat rozmery z dajakych dalsich suborov), lebo si neviem predstavit ako by som z rawka vycucal potrebne udaje.
Popripade, ak ste zastancom vsetkeho v jednom, teda ze by sa aj objemove data vykreslovali v tej istej apke ako plochy a sfera, tak by som to skusil dat dokopy a asi budem musiet premysliet svoje ify.
Dalej by som chcel spravit:
http://graphicsrunner.blogspot.sk/2009/01/volume-rendering-102-transfer-functions.html
-ak sa mi nepodaria oboje tak aspon tvorba vlastnej transferovej funkcie, potom by nemuselo byt az take zlozite tam dorobit aj vyklikavanie uzivatelom.
Rad by som to spravil aj so shadingom, tam to uz zacina vyzerat celkom zaujimavo.
Neviem ako to pojde, ale snad by som to mohol stihnut tento vikend a potom by som sa asi (ak nebudete mat ziadne postrehy || napady || pripomienky) zaoberal pisanim dajakeho textu, pretoze vcera som sa dozvedel od mojich kolegov, ze svocka je sice koncom aprila, ale text treba odovzdat o dva tyzdne (alebo take nieco). Ak su so svockou spojene este dajake ine veci co by som mal vykonat do urciteho terminu, dajte mi prosim vediet.
Dakuejm.

**Comment by RobertSpir** [comment]
Co sa tyka volume renderingu, tak budeme pouzivat vtk subory. Vecer vam
poslem jednak vstupny subor a aj funkcie do c# na nacitanie, lebo je to
pomerne netrivialny problem.

**Comment by RobertSpir** [comment]
hezke to je, len mi to stale vypisuje, ze transferfunc is not bind, ma to tak byt?
vtk som poslal na mail
Co sa tyka svk, tak do 21.4. treba odovzdat text, to su este 3 tyzdne (a staci 10 stran, a budeme mat vela obrazkov....) Ale mozete zacat rozmyslat nad nejakou osnovou.
Napriklad, najprv nejaky vseobecny pokec o potrebe vizualziacie vysledkov numerickych vypoctov, potom nieco o opengl a opentk kniznici a c#, o shaderoch. Ako je konkretne implementovane zobrazovanie na guli, ako je implementovane zobrazovanie na ploche. Potom nieco o volume renderingu a implementacii.

**Comment by xmatakt** [comment]
Hmm, tento problem som odstranil (nie idealne iba docasne), ale možno som
nahral nie najaktuálnejšiu verziu. Ked pridem domov pozriem sa co som
pokazil
Dňa 1.4.2016 19:54 používateľ "RobertSpir" notifications@github.com
napísal:

**Comment by xmatakt** [comment]
A mam do davat vsetko do jedneho?
Dňa 1.4.2016 20:16 používateľ "Timotej Maták" timotej.matak@gmail.com
napísal:

**Comment by xmatakt** [comment]
Neviem kde bol problem, ja som si stiahol to co som sem nahraval a ziadnu chybu mi nehlasil, ale aj tak som to prerobil tak, aby nemal ani najmensiu moznost hlasit chybu, takze ak ju stale hlasi tak netusim kde je problem.

**Comment by xmatakt** [comment]
Vykresluje aj bunky, ale su akesi roztiahnute.

**Comment by RobertSpir** [comment]
vyzera, ako keby vzdy ten objem roztiahol do kocky. Robi to aj ked som skusil zobrat len cast z tych raw volumov, tak to roztiahol

**Comment by xmatakt** [comment]
Hej, treba vzdy 3D texturu vytvarat rozmeru AxAxA, vcera som to zistil a
zabudol pushnut
Dňa 2.4.2016 8:41 používateľ "RobertSpir" notifications@github.com
napísal:

**Comment by RobertSpir** [comment]
aj volumerendering funguje v linuxe.
skuste spravit tie shadery nech su sucastou programu, cez properties->resources, bud ako string alebo ako file, aspon odpadnu vsetky problemy s cestami
v kode potom mozete k tym resources pristupovat cez VolumeRendering.Properties.Resources.nazov

**Comment by xmatakt** [comment]
Uz nacitava ako ste chceli, a vykresluje neroztiahnuto.
A sa to pokusit dat vsetko do jedneho programu, alebo volumerendering moze byt samozstatne?
Ja by som bol skor za samostatnost, keby som tam nahodov pridaval dajake ovladacie prvky, ale nedbam.

**Comment by xmatakt** [comment]
*mam sa... *nahodou

**Comment by RobertSpir** [comment]
moze to byt zatial samostatne. Skuste este spravit tu transfer function so shadingom, mozete to skusat aj na tych datach z toho blogu, z toho budu velmi pekne obrazky pre porotu na svk...

**Comment by xmatakt** [comment]
To mam v plane, obrazky su najdolezitejsie.

**Comment by RobertSpir** [comment]
presne tak, pripadne aj nejake video....

**Comment by xmatakt** [comment]
Ako vlastne prebieha svocka?, pridem s peknou farebnou prezentaciou a
nikoho nezaujima co je v praci ktoru som odovzdal? Komisia je nejaka
externa?
Dňa 2.4.2016 10:45 používateľ "RobertSpir" notifications@github.com
napísal:

**Comment by RobertSpir** [comment]
pracu dostane pred samotnou svockou nejaky oponent, ten si ju precita a napise posudok. Takze ta praca musi mat nejaku uroven :). (kto bude oponent este uvidim).
Komisia mava potom tusim 5 clenov, 2-3 z katedry, 1 z inej katedry, 1-2 z inej skoly (vacsinou Bock a Oksa). Vy tam odprezentujete pracu peknou, farebnou a dynamickou prezentaciou, pocas toho si komisia moze listovat v tej vasej praci. Nasledne oponent (alebo niekto z komisie, ak oponent chyba) precita posudok. Pripadne je kratka diskusia. (Bock sa casto zvykne pytat aj na nieco, co velmi nesuvisi s pracou...)
Najvacsiu vahu ma prezentacia, ale samozrejme aj samotny text ma vplyv, kedze oponent ho cita a robi posudok.

**Comment by xmatakt** [comment]
Cize mpm ma samostatnu kategoriu, nebudu tam pravi stavbari?
Dňa 2.4.2016 10:58 používateľ "RobertSpir" notifications@github.com
napísal:

**Comment by RobertSpir** [comment]
v podstate vacsina katedier ma vlastnu kategoriu, tusim 6 ucastnikov je potrebnych na samostatnu kategoriu. U nas je tento rok 9. Konkurencia bude hlavne vasi spoluziaci, jeden druhak a dvaja? piataci.
A podla toho, co sa zatial podarilo spravit vam a co mam informacie o niektorych ostatnych, tak to vidim celkom nadejne.

**Comment by xmatakt** [comment]
No neviem hrapkova a szeleova. Ale to je jedno ja nemam ktovieake ambicie,
len ma zaujimalo ci tam bedem cely den, ale ked su taketo kategorie tak to
je v pohode. Dakujem za info.
Dňa 2.4.2016 11:09 používateľ "RobertSpir" notifications@github.com
napísal:

**Comment by RobertSpir** [comment]
Zaciatok bude asi o osmej, kazdy bude mat cca 20 minut, takze koniec este pred obedom. Vyhlasenie vysledkov zvykne byt o piatej poobede pre vsetkych naraz v aule.
Nasledne je mozny postup do cesko-slovenskeho kola, tento rok v Brne

**Comment by xmatakt** [comment]
Tak, transferovu funkciu som rozbehal, aj ked vacsinu casu co som nad tym stravil som zabil tym, ze som sa snazil vyuzit kod pre kubicky spline z pocitacovej grafiky, aby som medzi jednotlivymi hodnotami interpoloval po kubickej krivke tak ako to robil clovek z blogu, ale z neznamych pricin aj ked sa javil moj spline ako funkcny (na grafike k nemu neboli ziadne pripomienky), nepodarilo sa mi generovat ziadne rozumne vysledky, tak som to zlinearnil.

**Comment by RobertSpir** [comment]
to uz je aj so shadingom? Ale nevyzera to tak pekne, ako na tom blogu. Pripada mi to, ako keby opacitu vobec nebral do uvahy. Na tej hlave vidno iba vonkajsiu vrstvu, ktora je nepriehladna
este spravte aby tie nacitane data boli v tom kockovom 3d objeme vycentrovane (napriklad tie bunky su teraz v hornej polovici a os otacania je tym padom mimo tych dat)

**Comment by xmatakt** [comment]
Nie to je iba transferova funkcia bez shadingu.

**Comment by xmatakt** [comment]
Neviem co ste pozerali, asi nie tu hlavu co ma on na blogu, lebo ta transferova funkcia je nasita na tie data, mam tam aj tu, ale ta sa zobrazuje 4x lebo neviem preco, tam je to trocha lepsie vidiet. Priehladnost funguje normalne, len musim dorobit ovladanie, aby si vsetky tie veci mohol navolit uzivatel sam, tam bude vidiet aspon co co robi, popripade co nefunguje. Pocitam, ze sa k tomu dostanem az vecer, ale dufam, ze to dnes zbucham.

**Comment by RobertSpir** [comment]
aha, ja som pozeral to co to nacita defaultne, tiez nejaka hlava, som nevedel, ze to je ina

**Comment by xmatakt** [comment]
jj aj to nacitavanie by som mal spravit nejako rozumne

**Comment by xmatakt** [comment]
Rozbehal som drivre v linuxe, uz to beha aj mne, aj ked ma akesi problemy s druhym monitorom ked je na druhehj grafike, ak je na intelackej nema problem ale zase nedokaze nacitat shadere...
V podstate by sa dalo povedat, ze to ide, ale nedaju sa nastavovat farby, nevyhadzuje colorDialogBox.

**Comment by RobertSpir** [comment]
zaujimave, ze ten colordialog nejde...ked som spravil samostatnu app len s jednym butónom, ktory vyvola kolordialog, tak to islo v pohode....mozno neregistruje kliknutie na tie veci

**Comment by xmatakt** [comment]
neviem, to by potom neregistroval nic vobec nic keby som klikol na
ktorykolvek ListView (su tam tri), nie som si isty ale tusim som vcera aj
menil nejake hodnoty cez ten horny, cize reagoval na klikanie. Aj ten label
cez ktory je vykreslena zvolena farba je nejaky pokazeny

2016-04-08 18:44 GMT+02:00 RobertSpir notifications@github.com:
